### PR TITLE
Tsify lib/clientstate-normalizer

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Depends on #4719

### CODE REVIEW

File: lib/clientstate-normalizer.ts [hunk 1]
```diff
@@ -1030,7 +1025,6 @@ export class ClientStateGoldenifier extends GoldenLayoutComponents {
                         tool.id,
                         tool.args,
                         tool.stdin,
-                        width,
```
Comment (partouf) [inline lib/clientstate-normalizer.ts:1033]: are you sure this is correct? Although it seems links with tools are currently broken, so it's hard to tell...

Comment (jeremy-rifkin) [reply lib/clientstate-normalizer.ts:1033]: Ah thank you, I think I did make a mistake here. I'm not 100% confident on anything here but going off the parameter names it should probably be
```ts
                    const stack = this.newToolStackFromCompiler(
                        session,
                        idxCompiler + 1,
                        tool.id,
                        tool.args,
                        tool.stdin,
                        width,
                    );
```

**Comment by partouf** [comment]
Made a separate issue for tool pane being broken if generated from clientstate https://github.com/compiler-explorer/compiler-explorer/issues/4726

File: lib/clientstate-normalizer.ts (hunk 1 above)
Comment (partouf) [reply lib/clientstate-normalizer.ts]: ohhh that might be why https://github.com/compiler-explorer/compiler-explorer/issues/4726 is broken !

**Pull request merged**